aten/src/ATen/cpu/vml.h

```diff
@@ -125,16 +126,42 @@ IMPLEMENT_VML_BUG(trunc)
 
 #if AT_MKL_ENABLED() && !defined(__APPLE__)
 
-#define IMPLEMENT_VML_MKL(op, mklop)                                         \
-  template <>                                                                 \
-  inline void v##op(float* out, const float* in, int64_t size) {              \
-    vms##mklop(size, in, out, VML_HA | VML_FTZDAZ_OFF | VML_ERRMODE_IGNORE); \
-  }                                                                           \
-  template <>                                                                 \
-  inline void v##op(double* out, const double* in, int64_t size) {            \
-    vmd##mklop(size, in, out, VML_HA | VML_FTZDAZ_OFF | VML_ERRMODE_IGNORE); \
+// NB: LP64 MKL is the most commonly used and thus we assume it here. That means
+// we need to expect MKL_INT to be of type int, which implies int32_t in most
+// cases.
+static_assert(
```

Comment:
ship this, but we should add a global setting to allow building / linking against MKL_INT=int64_t as well, it just has to be non-default. Send a patch to do that in a follow-up PR.